File: Entities/Dtos/WarehouseItemDto.cs
using Core.Entities;
using Core.Enums;

namespace Entities.Dtos
{
    public class WarehouseItemDto: IDto
    {
        public int WarehouseId { get; set; } = 1;
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string ColorName { get; set; }
        public ESize Size { get; set; } 
        public int Quantity { get; set; }
        public bool IsAvailableForSale { get; set; }
    }
}

File: UiPreparation/UI/src/app/core/components/user/warehouse/models/warehouse-item-dto.ts
import { ESize } from "../../../../enums/e-size.enum";

export interface WarehouseItemDto {
    warehouseId: number;
    productId: number;
    productName: string;
    colorName: string;
    size: ESize;
    quantity: number;
    isAvailableForSale: boolean;
}